client/java/transports-dataplex/src/main/java/io/openlineage/client/transports/dataplex/DataplexTransport.java

```diff
@@ -0,0 +1,132 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.client.transports.dataplex;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.rpc.FixedHeaderProvider;
+import com.google.auth.Credentials;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.cloud.datacatalog.lineage.v1.LineageSettings;
+import com.google.cloud.datacatalog.lineage.v1.ProcessOpenLineageRunEventRequest;
+import com.google.cloud.datalineage.producerclient.helpers.OpenLineageHelper;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageClient;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageProducerClient;
+import com.google.cloud.datalineage.producerclient.v1.SyncLineageProducerClientSettings;
+import com.google.protobuf.Struct;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineageClientException;
+import io.openlineage.client.OpenLineageClientUtils;
+import io.openlineage.client.transports.Transport;
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DataplexTransport extends Transport implements Closeable {
+
+  private final SyncLineageClient producerClient;
```

Comment:
@midamkina Should we also give the option to use the Async client?

Reply:
From an implementation point of view, one potential challenge about the Async client is that we would need a way to track the success/failure of the API call asynchronously as well.

Reply:
@jphalip done